extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -43,21 +44,31 @@ export class AmpExperiment extends AMP.BaseElement {
 
     const config = this.getConfig_();
     const results = Object.create(null);
+    const override = this.getVariantOverride_();
     const variants = Object.keys(config).map(experimentName => {
+      if (override[experimentName]) {
+        results[experimentName] = override[experimentName];
```

Comment:
Thanks for writing this up :-). I have thought about this. I feel it's not that risky because the very limited chars we allow here (0-9,a-z,-,_). 

And also, everything injected to the URL will be encoded first. So shouldn't be a big problem here.